numpy/lib/histograms.py

```diff
@@ -1070,11 +1072,19 @@ def histogramdd(sample, bins=10, range=None, normed=None, weights=None,
         dedges[i] = np.diff(edges[i])
 
     # Compute the bin number each sample falls into.
-    Ncount = tuple(
-        # avoid np.digitize to work around gh-11022
-        np.searchsorted(edges[i], sample[:, i], side='right')
-        for i in _range(D)
-    )
+    Ncount = D*[None]
+    for i in _range(D):
+        if uniform_binnings[i]:
+            # avoid np.searchsorted if dimension has uniform binning
+            left, right = edges[i][0], edges[i][-1]
+            nbins = len(edges[i]) - 1
+            ib = 1 + (nbins * (sample[:, i] - left) / (right - left))
```

Comment:
if right equals left(may happen when `edges=[[1]]`), an exception occurs.

Reply:
I wasn't able to trigger left==right, since it is protected by `_get_outer_edges()` earlier in the code, which checks for this pathological binning https://github.com/numpy/numpy/blob/6601776e7e1d84047f203e62ca5f3d2e43ca1da4/numpy/lib/histograms.py#L326-L329